zcash_keys/src/keys.rs

```diff
@@ -928,6 +928,351 @@ impl UnifiedFullViewingKey {
+            return Err(AddressGenerationError::ReceiverTypeNotSupported(
+                Typecode::Sapling,
+            ));
+
+            #[cfg(feature = "sapling")]
+            if let Some(extivk) = &self.sapling {
+                // If a Sapling receiver type is requested, we must be able to construct an
+                // address; if we're unable to do so, then no Unified Address exists at this
+                // diversifier and we use `?` to early-return from this method.
+                let diversifier = ::sapling::keys::Diversifier(_j.as_bytes().to_owned());
+                sapling = Some(
+                    extivk
+                        .address(diversifier)
+                        .ok_or(AddressGenerationError::InvalidSaplingDiversifierIndex(_j))?,
+                );
+            } else {
+                return Err(AddressGenerationError::KeyNotAvailable(Typecode::Sapling));
+            }
+        }
+
+        #[cfg(feature = "transparent-inputs")]
+        let mut transparent = None;
+        if _request.has_p2pkh {
+            #[cfg(not(feature = "transparent-inputs"))]
+            return Err(AddressGenerationError::ReceiverTypeNotSupported(
+                Typecode::P2pkh,
+            ));
+
+            #[cfg(feature = "transparent-inputs")]
+            if let Some(tivk) = self.transparent.as_ref() {
+                // If a transparent receiver type is requested, we must be able to construct an
+                // address; if we're unable to do so, then no Unified Address exists at this
+                // diversifier.
+                let transparent_j = to_transparent_child_index(_j)
+                    .ok_or(AddressGenerationError::InvalidTransparentChildIndex(_j))?;
+
+                transparent = Some(
+                    tivk.derive_address(transparent_j)
+                        .map_err(|_| AddressGenerationError::InvalidTransparentChildIndex(_j))?,
+                );
+            } else {
+                return Err(AddressGenerationError::KeyNotAvailable(Typecode::P2pkh));
+            }
+        }
+        #[cfg(not(feature = "transparent-inputs"))]
+        let transparent = None;
+
+        UnifiedAddress::from_receivers(
+            #[cfg(feature = "orchard")]
+            orchard,
+            #[cfg(feature = "sapling")]
+            sapling,
+            transparent,
+        )
+        .ok_or(AddressGenerationError::ShieldedReceiverRequired)
+    }
+
+    /// Searches the diversifier space starting at diversifier index `j` for one which will
+    /// produce a valid diversifier, and return the Unified Address constructed using that
+    /// diversifier along with the index at which the valid diversifier was found.
+    ///
+    /// Returns an `Err(AddressGenerationError)` if no valid diversifier exists or if the features
+    /// required to satisfy the unified address request are not properly enabled.
+    #[allow(unused_mut)]
+    pub fn find_address(
```

Comment:
Same thing here, modify `UnifiedFullViewingKey`'s impl to construct the uivk and delegate to this method.

Reply:
Great. That was the plan (see PR description).